### ISSUE TITLE
MA0042 should trigger for IAsyncDisposables

### ISSUE THREAD

**Issue by robv8r** [meziantou/Meziantou.Analyzer#1132]
### Version of the Meziantou.Analyzer NuGet package

3.0.63

### Rule Identifier

MA0042

### Target Framework

net10.0

### C# Language version

C# 14.0

### Description

As of version 3.0.63, the analyzer no longer reports MA0042 for `IAsyncDisposable` types, such as `Microsoft.Data.SqlClient`, version 7.0.1.

Prior to 3.0.63, the analyzer reports MA0042 as expected.

### Reproduction Steps

Minimal code:

```c#
using var connection = new SqlConnection(""); // 👈 As of version 3.0.63, MA0042 is not reported, but should be.
```

### Other information

```c#
using var connection = new SqlConnection(""); // 👈 In version 3.0.62, MA0042 is reported as expected.
```

Examples using versions 3.0.62 and 3.0.63 can be found here: https://github.com/robv8r/issue-ma0042

**Comment by meziantou** [comment]
I think this is expected. `SqlConnection` uses `DisposeAsync` from `DbConnection` (base class). This default implementation just call `Dipose()`, so it's not really async. It just add overhead.
A recent change detect this for DbConnection and DbCommand to reduce useless async migration. The logic is to check if the child class override the DisposeAsync method. If so, MA0042 is reported. Otherwise, no diagnostic as changing the code would simply create overhead and make the code more complex.

**Comment by robv8r** [comment]
Thanks for the quick response and justification.  A simple `F12` would have shown me exactly what you've described.

Sorry for the false alarm.

```c#
public virtual ValueTask DisposeAsync()
{
    Dispose();
    return default;
}
```

**Comment by robv8r** [comment]
Should the same apply to a `SqlDataReader`?  It calls `DbDataReader.DisposeAsync()` (base class).  The default implementation calls `Dispose()`.

```c#
// 👇 In version 3.0.70, MA0042 is reported
using var reader = await command.ExecuteReaderAsync(token).ConfigureAwait(false);
```
